sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -353,7 +371,88 @@ struct graphs_support;
 } // namespace device
 } // namespace info
 
-class node {};
+enum class dynamic_parameter_type {
+  usm,
+  accessor,
+  scalar
+};
+
+template <dynamic_parameter_type DynParamType>
+class dynamic_parameter;
+
+template <>
+class dynamic_parameter<dynamic_parameter_type::usm> {
+public:
+  dynamic_parameter(command_graph<graph_state::modifiable> graph,
+                    std::string label = "");
+
+  void register(void* param, node node);
+  void register(void* param);
```

Comment:
this API is designed to update the inputs&outputs (USM) of a sycl graph when they are changed in different iteration.  

Here is one possible example with pseudocode:
```
graph g;
g.begin_recording();
.... // workloads to be recorded
g.end_recoding();
exec_g = g.finalize()

// register all the possible inputs&outputs (suppose we have 3: p0, p1, p2) because we don't know which one will be changed
dynamic_parameter dp0(g);
dp0.register(p0);
dynamic_parameter dp1(g);
dp1.register(p1);
dynamic_parameter dp2(g);
dp2.register(p2);

// nothing is changed for the first iteration
submit(exec_g);

// p0 is changed to newP0
dp0.update(newP0, exec_g);
submit(exec_g);

// p0 is back, p1 and p2 are changed
dp0.update(p0, exec_g);
dp1.update(newP1, exec_g);
dp2.update(newP2, exec_g);
submit(exec_g);
```